fedoragsearch-transforms/islandora_transforms/slurp_all_MODS_to_solr.xslt

```diff
@@ -58,10 +59,9 @@
         </xsl:choose>
     </field>
 
-    <-- JIRA TRAC-876 Define utk_mods_etd_author_orcid in Solr -->
 
-    <xsl:if test="mods:name[@type='orcid']">
-        <xsl:variable name="orcidtrim" select="substring-after(mods:name[@valueURI],'http://orcid.org/')"/>
+    <xsl:if test="/mods:mods/mods:name[@authority='orcid']">
+        <xsl:variable name="orcidtrim" select="substring-after(/mods:mods/mods:name[@valueURI],'http://orcid.org/')"/>
```

Comment:
There's still an issue with the xsl:variable's select instruction: in the first argument of the substring-after() function, the actual value being selected is a `/mods:mods/mods:name` with an `@valueURI`, when really you're wanting `/mods:mods/mods:name/@valueURI`.  Here's a working example:

```xml
<xsl:if test="@authority='orcid'">
  <xsl:variable name="vORCID" select="substring-after(@valueURI, 'http://orcid.org/')"/>
  <field name="utk_mods_etd_author_orcid_s">
    <xsl:value-of select="$vORCID"/>
  </field>   
</xsl:if>
```
This is a slight modification on @markpbaggett's suggestion from yesterday. This keeps the current context (i.e. the `xsl:template match="..."` that is currently being used).

Reply:
@CanOfBees, sorry.  I meant to copy and paste that in.  I just missed that part in my review.  You're absolutely correct.